app/models/podcast.rb

```diff
@@ -66,7 +67,12 @@ def self.release!(options = {})
 
   def set_defaults
     self.explicit ||= "false"
-    self.link ||= embed_player_landing_url(self)
+  end
+
+  def set_empty_link
+    if link.nil?
```

Comment:
this should only update `link` if it's still blank after saving, whether it's a new podcast or updating an existing one.

Reply:
Minor nitpick: I might use `.blank?` here, to catch empty strings `""` as well?  Just in case the web forms submitted a blank string.